articles/contributing/style-guide.md

```diff
@@ -0,0 +1,451 @@
+
+- `ConstantArray`
+- `Head`
+- `LookupFunction`
+
+In particular, in almost all cases, we suggest using past participles where appropriate to indicate that a function name is strongly connected to an action or side effect elsewhere in a quantum program.
+For example,  `ControlledOnInt` uses the part participle form of the verb "control" to indicate that the function acts as an adjective to modify its argument.
+This name has the additional benefit of matching the semantics of the built-in `Controlled` functor, as discussed further below.
+Similarly, _agent nouns_ can be used to construct function and UDT names from operation names, as in the case of the name `Encoder` for a UDT that is strongly associated with `Encode`.
+
+In common cases, we suggest using particular nouns and adjectives:
+
+| Noun/Adjective | Expected Result |
+|----|----|
+| Claim | Unit, but fails if a condition is not true. (Contrast with `Assert`, which should be used for operations.)
+| Controlled | An operation that is a controlled version of an input operation. |
+| Identity | A UDT or operation that represents identity / no-op on some data type. |
+| Operation | An operation that is computed from another operation in some way. Use when no more specific case applies. |
+
+
+# [Guidance](#tab/guidance)
+
+We suggest:
+
+- Use verbs for operation names.
+- Use nouns or adjectives for function names.
+- Use nouns for user-defined types.
+- For all callable names, use `CamelCase` in strong preference to `pascalCase`, `snake_case`, or `ANGRY_CASE`.
+- Avoid the use of underscores `_` in function and operation names.
+
+# [Examples](#tab/examples)
+
+|   | Name | Description |
+|---|------|-------------|
+| ☑ | `operation ReflectAboutStart` | Clear use of a verb ("reflect") to indicate the effect of the operation. |
+| ☒ | <s>`operation XRotation`</s> | Use of noun phrase suggests function, rather than operation. |
+| ☒ | <s>`operation search_oracle`</s> | Use of `snake_case` contravenes Q# notation. |
+| ☑ | `function StatePreparationOracle` | Use of noun phrase suggests that the function returns an operation. |
+| ☑ | `function ClaimEqual` | Clear use of noun ("claim") to indicate that this is a function. |
+| ☒ | <s>`function GetRotationAngles`</s> | Use of verb ("get") suggests that this is an operation. |
+| ☑ | `newtype GeneratorTerm` | Use of noun phrase clearly refers to the result of calling the UDT constructor. |
+
+***
+
+### Shorthand and Abbreviations ###
+
+The above advice notwithstanding, there are many forms of shorthand that see common and pervasive use in quantum computing.
+We suggest using existing and common shorthand where it exists, especially for operations that are intrinsic to the operation of a target machine.
+For example, we choose the name `X` instead of `ApplyX`, and `Rz` instead of `RotateAboutZ`.
+When using such shorthand, operation names should be uppercase.
```

Comment:
I would argue they should always be upper case.

Reply:
Clarified, thanks!